src/components/message/threadAttachment/gitHubAttachment.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import React, { useState, useEffect } from 'react';
+import path from 'path';
+// $FlowFixMe
+import parseGithubUrl from 'parse-github-url';
+import { Loading } from 'src/components/loading';
+import { Container, Column, GitHubBadge } from './style';
+import { ThreadTitle } from 'src/components/inboxThread/style';
+import { InnerMessageContainer } from '../style';
+
+const GitHubAttachment = (props: { url: string }) => {
+  const { url } = props;
+  const [loading, setLoading] = useState(true);
+  const [apiData, setApiData] = useState(null);
+  const [urlData] = useState(parseGithubUrl(url));
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const apiUrl = path.join('https://api.github.com/repos/', urlData.path);
```

Comment:
Are we going to hit rate limits here?

Reply:
>  For API requests using Basic Authentication or OAuth, you can make up to 5000 requests per hour. Authenticated requests are associated with the authenticated user, regardless of whether Basic Authentication or an OAuth token was used. This means that all OAuth applications authorized by a user share the same quota of 5000 requests per hour when they authenticate with different tokens owned by the same user.
> For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the **originating IP address**, and not the user making requests.

That seems like a bit of a problem any suggestions on an acceptable workaround?

Reply:
Is there a way to access Oauth tokens for Github authenticated users? If this is possible then I can keep the solution the way it is and add another fallback,  or I can just remove the api request altogether and  render as follows. ![Imgur](https://i.imgur.com/oUKL5hI.png)